src/core/positioning/tcpreceiver.cpp

```diff
@@ -65,44 +65,38 @@ void TcpReceiver::handleDisconnectDevice()
 
 void TcpReceiver::setSocketState( const QAbstractSocket::SocketState socketState )
 {
-  if ( mSocketState == socketState )
-  {
-    return;
-  }
+  emit socketStateChanged( socketState );
+  emit socketStateStringChanged( socketStateString() );
```

Comment:
like for egeniouss, could be replaced by a direct connection of the socket stateChanged with abstracGnss::stateChanged